components/ledger/internal/storage/ledgerstore/balances.go

```diff
@@ -42,9 +42,13 @@ func (store *Store) GetAggregatedBalances(ctx context.Context, q GetAggregatedBa
 					return "", nil, newErrInvalidQuery("'metadata' column can only be used with $match")
 				}
 				match := metadataRegex.FindAllStringSubmatch(key, 3)
-				needJoinMetadata = true
+				needMetadata = true
```

Comment:
never putted to false ? We always need the metadata ?

Reply:
Default is false